custom_components/elasticsearch/es_index_manager.py

```diff
@@ -194,55 +219,48 @@ async def _create_legacy_template(self):
-            LOGGER.debug("Ensuring ILM Policy is attached to existing index")
-            try:
-                await client.indices.put_settings(
-                    index=self.index_alias,
-                    preserve_existing=True,
-                    body={
-                        "index.lifecycle.name": self._ilm_policy_name,
-                        "index.lifecycle.rollover_alias": self.index_alias,
-                    },
-                )
-            except ElasticsearchException as err:
-                LOGGER.exception("Error updating index ILM settings: %s", err)
 
-    async def _create_ilm_policy(self, config):
+    async def _create_basic_ilm_policy(self, ilm_policy_name):
         """Create the index lifecycle management policy."""
         from elasticsearch7.exceptions import TransportError
 
         client = self._gateway.get_client()
 
         try:
-            existing_policy = await client.ilm.get_lifecycle(self._ilm_policy_name)
+            existing_policy = await client.ilm.get_lifecycle(ilm_policy_name)
         except TransportError as err:
             if err.status_code == 404:
                 existing_policy = None
             else:
                 LOGGER.exception("Error checking for existing ILM policy: %s", err)
                 raise err
 
-        ilm_hot_conditions = {"max_size": config.get(CONF_ILM_MAX_SIZE)}
+        if existing_policy:
+            LOGGER.info("Found existing ILM Policy, do nothing '%s'", ilm_policy_name)
+            return
 
         policy = {
             "policy": {
                 "phases": {
                     "hot": {
                         "min_age": "0ms",
-                        "actions": {"rollover": ilm_hot_conditions},
-                    },
-                    "delete": {
-                        "min_age": config.get(CONF_ILM_DELETE_AFTER),
-                        "actions": {"delete": {}},
+                        "actions": {
+                            "rollover": {
+                                "max_age": "30d",
+                            },
+                        },
                     },
+                    "delete": {"min_age": "365d", "actions": {"delete": {}}},
                 }
             }
         }
 
-        if existing_policy:
-            LOGGER.info("Updating existing ILM Policy '%s'", self._ilm_policy_name)
-        else:
-            LOGGER.info("Creating ILM Policy '%s'", self._ilm_policy_name)
+        if self._gateway.es_version.supports_max_primary_shard_size():
+            LOGGER.debug("Elasticsearch supports max_primary_shard_size, including in ILM template.")
+            policy["policy"]["phases"]["hot"]["actions"]["rollover"][
+                "max_primary_shard_size"
+            ] = "50gb"
+
+        LOGGER.info("Creating ILM Policy '%s'", ilm_policy_name)
 
-        await client.ilm.put_lifecycle(self._ilm_policy_name, policy)
+        await client.ilm.put_lifecycle(ilm_policy_name, policy)
```

Comment:
Also not a part of this PR...but can this be wrapped in a try/catch, so we can add our own exception logging and raise our own error?